exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java

```diff
@@ -207,6 +209,28 @@ private void javaPropertiesToSystemOptions() {
   }
 
   /**
+   * Disallow registering drillbit when:
+   * 1. version is unknown;
+   * 2. drillbit with different version has been already registered.
+   */
+  private void checkVersion(DrillbitEndpoint endpoint) throws DrillbitStartupException {
+    String currentVersion = endpoint.getVersion();
+    if (DrillVersionInfo.UNKNOWN_VERSION.equals(currentVersion)) {
+      throw new DrillbitStartupException("Drillbit version is unknown.");
+    }
+
+    for (DrillbitEndpoint registeredEnpoint : coord.getAvailableEndpoints()) {
+      if (!currentVersion.equals(registeredEnpoint.getVersion())) {
```

Comment:
How does this check work with regards to [protobuf backwards compatibility](https://developers.google.com/protocol-buffers/docs/javatutorial#extending-a-protocol-buffer)? Endpoints currently do not have a version. So if two bits (A and B), one with and one without this change, are started, the two start order cases (A after B, and B after A) should pass.